x-pack/plugins/cases/public/plugin.ts

```diff
@@ -115,6 +115,7 @@ export class CasesUiPlugin
     const getCasesContext = getCasesContextLazy({
       externalReferenceAttachmentTypeRegistry: this.externalReferenceAttachmentTypeRegistry,
       persistableStateAttachmentTypeRegistry: this.persistableStateAttachmentTypeRegistry,
+      getFilesClient: plugins.files.filesClientFactory.asScoped,
```

Comment:
Not for the PR but something that Antonio and I discussed was a way to avoid having to update each of these places whenever we have to add a new property (files client, attachment registries, etc). One thought I had was to pass a function that would return all these instead.

Do you have any ideas @cnasikas ?

Reply:
Yeah, this because all the `get...CasesLazy` take the same params.

So this:

  ```
    ui: {
        getCases: (props) =>
          getCasesLazy({
            ...props,
            externalReferenceAttachmentTypeRegistry: this.externalReferenceAttachmentTypeRegistry,
            persistableStateAttachmentTypeRegistry: this.persistableStateAttachmentTypeRegistry,
            getFilesClient: plugins.files.filesClientFactory.asScoped,
          }),
        getCasesContext,
        getRecentCases: (props) =>
          getRecentCasesLazy({
            ...props,
            externalReferenceAttachmentTypeRegistry: this.externalReferenceAttachmentTypeRegistry,
            persistableStateAttachmentTypeRegistry: this.persistableStateAttachmentTypeRegistry,
            getFilesClient: plugins.files.filesClientFactory.asScoped,
          }),
        // @deprecated Please use the hook getUseCasesAddToNewCaseFlyout
        getCreateCaseFlyout: (props) =>
          getCreateCaseFlyoutLazy({
            ...props,
            externalReferenceAttachmentTypeRegistry: this.externalReferenceAttachmentTypeRegistry,
            persistableStateAttachmentTypeRegistry: this.persistableStateAttachmentTypeRegistry,
            getFilesClient: plugins.files.filesClientFactory.asScoped,
          }),
        // @deprecated Please use the hook getUseCasesAddToExistingCaseModal
        getAllCasesSelectorModal: (props) =>
          getAllCasesSelectorModalLazy({
            ...props,
            externalReferenceAttachmentTypeRegistry: this.externalReferenceAttachmentTypeRegistry,
            persistableStateAttachmentTypeRegistry: this.persistableStateAttachmentTypeRegistry,
            getFilesClient: plugins.files.filesClientFactory.asScoped,
          }),
      },
```

could be something like

 ```
   const getLazyParams = (props) => ({
      ...props,
      externalReferenceAttachmentTypeRegistry: this.externalReferenceAttachmentTypeRegistry,
      persistableStateAttachmentTypeRegistry: this.persistableStateAttachmentTypeRegistry,
      getFilesClient: plugins.files.filesClientFactory.asScoped,
    });

    return {
      api: createClientAPI({ http: core.http }),
      ui: {
        getCases: (props) => getCasesLazy(getLazyParams(props)),
        getCasesContext,
        getRecentCases: (props) => getRecentCasesLazy(getLazyParams(props)),
        // @deprecated Please use the hook getUseCasesAddToNewCaseFlyout
        getCreateCaseFlyout: (props) => getCreateCaseFlyoutLazy(getLazyParams(props)),
        // @deprecated Please use the hook getUseCasesAddToExistingCaseModal
        getAllCasesSelectorModal: (props) => getAllCasesSelectorModalLazy(getLazyParams(props)),
      },
```

This popped up because we were discussing how the `getFilesClient` had to be added in a few places and that process was a bit error-prone.

Reply:
Make sense to me and it seems cleaner.

Reply:
Ok cool, I'll create a ticket for this, no need to do it in this PR.

Reply:
https://github.com/elastic/kibana/issues/153546

Reply:
I was just going to do it 😆

Reply:
Haha, up to you, maybe not in this PR? It has plenty of changes already :)

Reply:
Yes please not in this PR 🙂